packages/kbn-monaco/src/esql/language.ts

```diff
@@ -107,6 +110,40 @@ export const ESQLLang: CustomLangModuleType<ESQLCallbacks> = {
           suggestions: wrapAsMonacoSuggestions(suggestions),
         };
       },
+      async resolveCompletionItem(item, token): Promise<monaco.languages.CompletionItem> {
+        if (!callbacks?.getFieldsMetadata) return item;
+
+        const fieldsMetadataClient = await callbacks?.getFieldsMetadata();
+
+        if (
+          !fieldsMetadataClient ||
+          // If item is not a ECS field, no need to fetch metadata
+          item.kind !== 4 ||
+          // We set sortText to '1D' for ECS fields, so if not ECS, no need to fetch description
+          item.sortText !== '1D' ||
```

Comment:
This check makes me a bit wary. The particular sort text we use could easily change and I don't know that it would be clear that this check is broken.

A few wild ideas
- We've already loaded the full list of ECS field names at this point. Could we access that here somehow and check the field name against the list?
- Would there be much of a performance impact if we always made the ECS metadata call?
- We could import this sort text as a constant in both places it is used, making the link very clear.
- We could try violating the `CompletionItem` interface by adding a property to indicate that this is an ECS field. It's possible Monaco wouldn't remove it and we could check for that. Don't love this because it feels hacky, but...

Reply:
I like the constant for the sort text idea. Very clear and simple. We can access the cache but that would require circular cross-package imports. And if we can not fetch the metadata call request if not needed then I think that's a win 🎊

Reply:
I don't mind the constant sort text idea for a quick fix. However, I do expect the system we use to sort things to change quite a bit in the future. There are many design asks for fine-grained sorting control in various contexts. I think we will have to evolve beyond what we have today and possibly assign sort-texts based on a set of rules or a priority graph.

> We can access the cache as well
I think this would be the best solution if it isn't too hard. Just look out for escaped field names.

Reply:
@drewdaemon You're right I ended up using the cache instead for clarity & more explicit check. I also added a constant for the sort as I thought it's good to have the constant anyway. Changes are here [`5f0d286` (#192646)](https://github.com/elastic/kibana/pull/192646/commits/5f0d2868e449979ce4732c1d5385b683d8315c87)

Reply:
Thank you @qn895 !